opentelekomcloud/resource_opentelekomcloud_kms_key_v1.go

```diff
@@ -0,0 +1,287 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"key_alias": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"key_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"key_description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"realm": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"domain_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"creation_date": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"scheduled_deletion_date": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"is_enabled": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"default_key_flag": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expiration_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"origin": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"pending_days": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceKmsKeyV1Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	kmsKeyV1Client, err := config.kmsKeyV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack kms key client: %s", err)
+	}
+
+	createOpts := &keys.CreateOpts{
+		KeyAlias:       d.Get("key_alias").(string),
+		KeyDescription: d.Get("key_description").(string),
+		Realm:          d.Get("realm").(string),
+	}
+
+	log.Printf("[DEBUG] Create Options: %#v", createOpts)
+	v, err := keys.Create(kmsKeyV1Client, createOpts).ExtractKeyInfo()
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack key: %s", err)
+	}
+	log.Printf("[INFO] Key ID: %s", v.KeyID)
+
+	// Wait for the key to become enabled.
+	log.Printf("[DEBUG] Waiting for key (%s) to become enabled", v.KeyID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{WaitingForEnableState, DisabledState},
+		Target:     []string{EnabledState},
+		Refresh:    KeyV1StateRefreshFunc(kmsKeyV1Client, v.KeyID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for key (%s) to become ready: %s",
+			v.KeyID, err)
+	}
+
+	if !d.Get("is_enabled").(bool) {
+		key, err := keys.DisableKey(kmsKeyV1Client, v.KeyID).ExtractKeyInfo()
+		if err != nil {
+			return fmt.Errorf("Error disabling key: %s.", err)
+		}
+
+		if key.KeyState != DisabledState {
+			return fmt.Errorf("Error disabling key, the key state is: %s", key.KeyState)
+		}
+	}
+
+	// Store the key ID now
+	d.SetId(v.KeyID)
+	d.Set("key_id", v.KeyID)
```

Comment:
You don't need to do this if you are already using the `SetId`, just store this in one place.